FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,47 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, response)):
+				if let items = try? FeedItemsMapper.map(data, response) {
+					completion(.success(items))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
`FeedItemsMapper.map` is an internal detail of `RemoteFeedLoader` and we know that it can throw only `RemoteFeedLoader.Error.invalidData`. Current implementation is good. At the same time there is an implicit duplication of error handling. `try` here can be caught and thus error will be defined in one place that is `FeedItemsMapper.map`

Please check the suggestion.

```suggestion
				do {
					let items = try FeedItemsMapper.map(data, response)
					completion(.success(items))
				} catch {
					completion(.failure(error))
				}
```